### PR TITLE
PTQ ONNX Update requirements

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Update version of packages to the latest for onnxruntime-openvino, onnx, skl2onnx.

### Reason for changes

Updating a version of onnxruntime-openvino should fix the several problems with inferencing of quantized models.
There are shufflenet's family models and yolo's family.

### Related tickets

91534

### Tests

No tests

### CODE REVIEW

File: nncf/experimental/onnx/graph/onnx_graph.py [hunk 1]
```diff
@@ -285,9 +285,9 @@ def get_edge_shape(self, edge_name: str) -> List[int]:
         self._update_activation_tensors(do_shape_inference=True)
         if edge_name in self._activations_tensor_name_to_value_info:
             return ONNXGraph.get_tensor_shape(self._activations_tensor_name_to_value_info[edge_name])
-        raise RuntimeError('There is no edge with the name {}'.format(edge_name))
+        return None
```
Comment (KodiaqQ) [inline nncf/experimental/onnx/graph/onnx_graph.py:288]: Why it returns `None` now? Is it checked somehow in the code that uses this method? Can't we get unpredictable behavior during quantization?

File: nncf/experimental/onnx/graph/onnx_graph.py [hunk 2]
```diff
@@ -304,13 +304,16 @@ def get_edge_dtype(self, edge_name: str) -> int:
         self._update_activation_tensors(do_shape_inference=True)
         if edge_name in self._activations_tensor_name_to_value_info:
             return self._activations_tensor_name_to_value_info[edge_name].type.tensor_type.elem_type
-        raise RuntimeError('There is no edge with the name {}'.format(edge_name))
+        return None
```
Comment (KodiaqQ) [inline nncf/experimental/onnx/graph/onnx_graph.py:307]: Same.

**Review by alexsu52 — APPROVED** [review]
LGTM

File: nncf/experimental/onnx/graph/onnx_graph.py (hunk 1 above)
Comment (kshpv) [reply nncf/experimental/onnx/graph/onnx_graph.py]: I revert these changes, we can discuss later about returning None or raising the Exception during refactoring ONNXGraph

**Review by KodiaqQ — APPROVED** [review]

File: tests/onnx/quantization/test_classification_models_graph.py [hunk 3]
```diff
@@ -41,16 +40,18 @@
...
-    torch.onnx.export(model, x, onnx_model_path, opset_version=13)
+    # Ticket 96177
+    torch.onnx.export(model, x, onnx_model_path, opset_version=13, training=torch.onnx.TrainingMode.TRAINING)
 
     original_model = onnx.load(onnx_model_path)
     quantized_model = min_max_quantize_model(model_to_test.input_shape, original_model)
     compare_nncf_graph(quantized_model, model_to_test.path_ref_graph)
-    infer_model(model_to_test.input_shape, quantized_model)
+    # Ticket 96177
+    # infer_model(model_to_test.input_shape, quantized_model)
```
Comment (andrey-churkin) [inline tests/onnx/quantization/test_classification_models_graph.py:57]: Commented code is not allowed. Could you please remove it if it is not needed now?

File: tests/onnx/sanity_tests/test_classification_sanity_sample.py [hunk 4]
```diff
@@ -82,16 +79,20 @@ def mock_dataset_creator(dataset_path, input_name, input_shape, batch_size, shuf
...
-                               'OpenVINOExecutionProvider'])
-    _input = np.random.random(input_shape)
-    input_name = sess.get_inputs()[0].name
-    _ = sess.run([], {input_name: _input.astype(np.float32)})
+    # Ticket 96177
+    # sess = rt.InferenceSession(onnx_output_model_path, providers=[
+    #                            'OpenVINOExecutionProvider'])
+    # _input = np.random.random(input_shape)
+    # input_name = sess.get_inputs()[0].name
+    # _ = sess.run([], {input_name: _input.astype(np.float32)})
```
Comment (andrey-churkin) [inline tests/onnx/sanity_tests/test_classification_sanity_sample.py:98]: Commented code is not allowed. Could you please remove it if it is not needed now?

**Review by andrey-churkin — CHANGES_REQUESTED** [review]

File: setup.py [hunk 5]
```diff
@@ -119,14 +119,14 @@ def find_version(*file_paths):
         "setuptools==59.5.0"
     ],
     "onnx": [
-        "torch==1.9.1",
-        "torchvision==0.10.1",
-        "onnx==1.11.0",
+        "torch==1.13.0",
```
Comment (vshampor) [inline setup.py:122]: We need torch 1.12 to be aligned with the PT backend.

File: nncf/experimental/onnx/graph/nncf_graph_builder.py [hunk 6]
```diff
@@ -195,7 +195,12 @@ def create_nncf_graph(onnx_model: ModelProto) -> NNCFGraph:
                 tensor_shape = GraphConverter._get_tensor_shape(onnx_graph, output_edge)
 
                 output_node_id = nncf_graph.get_node_by_name(output_node.name).node_id
-                onnx_dtype = onnx_graph.get_edge_dtype_name(output_edge)
+                try:
+                    onnx_dtype = onnx_graph.get_edge_dtype_name(output_edge)
+                except RuntimeError:
+                    # If this edge was not added during inference of ONNX model,
+                    # we do not add it to NNCFGraph
```
Comment (vshampor) [inline nncf/experimental/onnx/graph/nncf_graph_builder.py:202]: Add explanation on how this manifests itself for BatchNorm layers exported in training mode.

File: tests/onnx/data/reference_graphs/original_nncf_graph/vgg16.dot [hunk 7]
```diff
@@ -1,81 +1,101 @@
 strict digraph  {
-"0 Conv_0" [id=0, type=Conv];
```
Comment (vshampor) [inline tests/onnx/data/reference_graphs/original_nncf_graph/vgg16.dot:2]: Consider expanding test cases to cover both training-mode and eval-mode input models. Separate the reference files accordingly.

**Review by vshampor — CHANGES_REQUESTED** [review]

File: setup.py (hunk 5 above)
Comment (kshpv) [reply setup.py:122]: Why should we align with PT backend here?

File: tests/onnx/data/reference_graphs/original_nncf_graph/vgg16.dot (hunk 7 above)
Comment (kshpv) [reply tests/onnx/data/reference_graphs/original_nncf_graph/vgg16.dot:2]: I don't think that we should expand model covering by adding train mode as OpenVINOExecutionProvider doesn't support inference of some layers exported in training mode, e.g. BatchNorm

File: tests/onnx/quantization/test_classification_models_graph.py (hunk 3 above)
Comment (kshpv) [reply tests/onnx/quantization/test_classification_models_graph.py:57]: This will be reverted back after the ticket 96177 will be implemented, because of that I commented this

File: setup.py (hunk 5 above)
Comment (kshpv) [reply setup.py:122]: after offline discussion, we decided to downgrade torch to 1.12.1

Comment (kshpv) [reply setup.py]: Done

File: tests/onnx/quantization/test_classification_models_graph.py (hunk 3 above)
Comment (kshpv) [reply tests/onnx/quantization/test_classification_models_graph.py:61]: Dicussed with @vshampor, he suggested to add explicit comments and it would be enough

**Review by vshampor — APPROVED** [review]

**Review by andrey-churkin — APPROVED** [review]
